Docs build fail

Building docs currently fails with
```
Extension error (sphinx.ext.napoleon):
Handler <function _process_docstring at 0x7ff3b502fea0> for event 'autodoc-process-docstring' threw an exception (exception: pop from an empty deque)
make: Leaving directory '/home/runner/work/intelmq/intelmq/docs'
make: *** [Makefile:22: html] Error 2
Error: Process completed with exit code 2.
```

upstream issues:
- https://github.com/sphinx-doc/sphinx/issues/10613
- https://github.com/sphinx-doc/sphinx/issues/10701

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -98,6 +98,7 @@ CHANGELOG
 ### Documentation
 - Feeds: Add documentation for newly supported dataplane feeds, see above (PR#2102 by Mikk Margus Möll).
 - Installation: Restructured the whole document to make it clearer and straight-forward (PR#2113 by Sebastian Wagner).
+- Add workaround for https://github.com/sphinx-doc/sphinx/issues/10701 (PR#2225 by Sebastian Wagner, kudos @yarikoptic, fixes #2224).
 
 ### Packaging
 

--- a/docs/conf.py
+++ b/docs/conf.py
@@ -116,3 +116,25 @@ def run_autogen(_):
 def setup(app):
     app.connect("builder-inited", run_apidoc)
     app.connect("builder-inited", run_autogen)
+
+
+
+import sphinx
+if sphinx.__version__ == '5.1.0':
+    # see https://github.com/sphinx-doc/sphinx/issues/10701
+    # workaround copied from https://github.com/datalad/datalad/pull/6883
+
+    # Although crash happens within NumpyDocstring, it is subclass of GoogleDocstring
+    # so we need to overload method there
+    from sphinx.ext.napoleon.docstring import GoogleDocstring
+    from functools import wraps
+
+
+    @wraps(GoogleDocstring._consume_inline_attribute)
+    def _consume_inline_attribute_safe(self):
+        try:
+            return self._consume_inline_attribute_safe()
+        except:
+            return "", []
+
+    GoogleDocstring._consume_inline_attribute = _consume_inline_attribute_safe

